### PR TITLE
Add observer to CommandBarItem menu

### PR DESCRIPTION
**Pull request by joannaquu**
### Platforms Impacted
- [x] iOS
- [ ] macOS

### Description of changes

Binary change:

Total increase: 2,688 bytes
Total decrease: 0 bytes
| File | Before | After | Delta |
|------|-------:|------:|------:|
| Total | 29,800,000 bytes | 29,802,688 bytes | ⚠️ 2,688 bytes |
<details>
<summary> Full breakdown </summary>

| File | Before | After | Delta |
|------|-------:|------:|------:|
| CommandBarItem.o | 113,600 bytes | 115,728 bytes | ⚠️ 2,128 bytes |
| __.SYMDEF | 4,557,712 bytes | 4,558,096 bytes | ⚠️ 384 bytes |
| CommandBarButton.o | 114,040 bytes | 114,216 bytes | ⚠️ 176 bytes |
</details>

Changes on `CommandBarItem` objects should automatically trigger updates to their corresponding `CommandBarButtons`. Currently, this is not the case for `menu` so updates are not propagated. This PR adds an observer to `menu` and updates it in the `updateState` function.

### Verification

<details>
<summary>Visual Verification</summary>

| Before                                       | After                                      |
|----------------------------------------------|--------------------------------------------|
| Simulator Screen Recording - iPhone 14 Pro - 2023-03-03 at 13 22 33 | Simulator Screen Recording - iPhone 14 Pro - 2023-03-03 at 13 23 29 |
</details>

### Pull request checklist

This PR has considered:
- [ ] Light and Dark appearances
- [ ] iOS supported versions (all major versions greater than or equal current target deployment version)
- [ ] VoiceOver and Keyboard Accessibility
- [ ] Internationalization and Right to Left layouts
- [ ] Different resolutions (1x, 2x, 3x)
- [ ] Size classes and window sizes (iPhone vs iPad, notched devices, multitasking, different window sizes, etc)
- [ ] iPad [Pointer interaction](https://developer.apple.com/documentation/uikit/pointer_interactions)
- [ ] [SwiftUI](https://developer.apple.com/tutorials/swiftui) consumption (validation or new demo scenarios needed)
- [ ] Objective-C exposure (provide it only if needed)
 ###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/fluentui-apple/pull/1623)

### CODE REVIEW

File: ios/FluentUI/Command Bar/CommandBarItem.swift [hunk 1]
```diff
@@ -148,7 +148,13 @@ open class CommandBarItem: NSObject {
 	/// property will disable the `itemTappedHandler`. The customControlView must manually handle gesutres.
     @objc public var customControlView: (() -> UIView)?
 
-    @objc public lazy var menu: UIMenu? = nil // Only lazy property can be used with @available
+    @objc public lazy var menu: UIMenu? = nil { // Only lazy property can be used with @available
+        didSet {
+            if menu != oldValue {
+                propertyChangedUpdateBlock?(self, false)
```
Comment (harrieshin) [inline ios/FluentUI/Command Bar/CommandBarItem.swift:154]: nit: can we also comment what the false is for with /*shouldUpdateGroupState*/

**Comment by harrieshin** [comment]
can we add a demoapp scenario where we actually set the menu item?  
						let menuItems =  @ [
							UIAction(title:"foo",
									 image: " bar"
									 handler: nil) ]


						commandItem.menu = UIMenu(children: menuItems)

**Comment by joannaquu** [comment]
> can we add a demoapp scenario where we actually set the menu item?

We currently have a menu in the demo app already. Would you like another one?
```
let copyItem = itemGroups[4][0]
copyItem.menu = UIMenu(children: [UIAction(title: "Copy Image", image: UIImage(named: "copy24Regular"), handler: { _ in }),
                                  UIAction(title: "Copy Text", image: UIImage(named: "text24Regular"), handler: { _ in })])
```

**Comment by harrieshin** [comment]
ah ok i didn't know it already existing. then we should have a before and after case where it does test updating menu after it is set?

**Review by harrieshin — APPROVED** [review]

**Pull request merged**